mlx/backend/common/matmul.h

```diff
@@ -0,0 +1,78 @@
+// Copyright © 2025 Apple Inc.
+
+#pragma once
```

Comment:
Maybe we should move this to `gpu/matmul.h` and include the `check_transpose` function there?

Reply:
Can cpu backend benefit from `collapse_batches`? It is not used there but I think it might help someday.